### ISSUE TITLE
SSH connection failed: None of the prompts looked like password prompts

### ISSUE THREAD

**Issue by latota007** [ansible/pylibssh#348]
##### SUMMARY
Can´t connect to a Cisco Device and you get a "ssh connection failed: None of the prompts looked like password prompts: [b'Clave:']" error. Our User and Password prompt in Cisco Devices is in Spanish. Like this:
"Usuario:" (User)
"Clave:" (Password)
The problem is random. Sometimes fails and sometimes not and its happend in random devices too.

##### ISSUE TYPE
- Bug Report

##### PYLISSH and LIBSSH VERSION
```
"libssh.x86_64  0.9.6-3.el8"
"ansible-pylibssh 0.4.0"
```

##### OS / ENVIRONMENT
This is running on AWX 21.2.0  with custom EE based on "quay.io/ansible/ansible-runner:latest" over CentOS Stream release 8 docker image
The targets are Cisco Catalyst 9800-40 and Catalyst 9800-80 running IOS-XE 17.3.4

##### STEPS TO REPRODUCE
Running this task

```
- name: Verifico si el SSID Invigal existe en el WLC
   ios_command:
     commands: 'show running-config | include wlan "Invigal Wlan Profile" 3 Invigal'
   register: resultado
```
##### EXPECTED RESULTS
When run the job you get devices that finished succesfully and another not.

```
TASK [Verifico si el SSID Invigal existe en el WLC] ****************************
ok: [wlc_campus_butty275.bancogalicia.com.ar] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "commands": [
                "show running-config | include wlan \\"Invigal Wlan Profile\\" 3 Invigal"
            ],
            "interval": 1,
            "match": "all",
            "provider": null,
            "retries": 10,
            "wait_for": null
        }
    },
    "stdout": [
        "wlan \\"Invigal Wlan Profile\\" 3 Invigal"
    ],
    "stdout_lines": [
        [
            "wlan \\"Invigal Wlan Profile\\" 3 Invigal"
        ]
    ]
}

```

##### ACTUAL RESULTS
This is the debug output for a failed device in the same job.

```
TASK [Verifico si el SSID Invigal existe en el WLC] ****************************
<wlc_campus_ctes415.bancogalicia.com.ar> attempting to start connection
<wlc_campus_ctes415.bancogalicia.com.ar> using connection plugin ansible.netcommon.network_cli
<wlc_campus_ctes415.bancogalicia.com.ar> local domain socket does not exist, starting it
<wlc_campus_ctes415.bancogalicia.com.ar> control socket path is /home/runner/.ansible/pc/3a352efa29
<wlc_campus_ctes415.bancogalicia.com.ar> redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
<wlc_campus_ctes415.bancogalicia.com.ar> Loading collection ansible.netcommon from /usr/share/ansible/collections/ansible_collections/ansible/netcommon
<wlc_campus_ctes415.bancogalicia.com.ar> redirecting (type: terminal) ansible.builtin.ios to cisco.ios.ios
<wlc_campus_ctes415.bancogalicia.com.ar> Loading collection cisco.ios from /usr/share/ansible/collections/ansible_collections/cisco/ios
<wlc_campus_ctes415.bancogalicia.com.ar> redirecting (type: cliconf) ansible.builtin.ios to cisco.ios.ios
<wlc_campus_ctes415.bancogalicia.com.ar> local domain socket listeners started successfully
<wlc_campus_ctes415.bancogalicia.com.ar> loaded cliconf plugin ansible_collections.cisco.ios.plugins.cliconf.ios from path /usr/share/ansible/collections/ansible_collections/cisco/ios/plugins/cliconf/ios.py for network_os ios
<wlc_campus_ctes415.bancogalicia.com.ar> ssh type is set to auto
<wlc_campus_ctes415.bancogalicia.com.ar> autodetecting ssh_type
<wlc_campus_ctes415.bancogalicia.com.ar> ssh type is now set to libssh
<wlc_campus_ctes415.bancogalicia.com.ar> local domain socket path is /home/runner/.ansible/pc/3a352efa29
<wlc_campus_ctes415.bancogalicia.com.ar> Using network group action ios for ios_command
<wlc_campus_ctes415.bancogalicia.com.ar> ANSIBLE_NETWORK_IMPORT_MODULES: enabled
<wlc_campus_ctes415.bancogalicia.com.ar> ANSIBLE_NETWORK_IMPORT_MODULES: found ios_command  at /usr/share/ansible/collections/ansible_collections/cisco/ios/plugins/modules/ios_command.py
<wlc_campus_ctes415.bancogalicia.com.ar> ANSIBLE_NETWORK_IMPORT_MODULES: running ios_command
<wlc_campus_ctes415.bancogalicia.com.ar> ANSIBLE_NETWORK_IMPORT_MODULES: complete
The full traceback is:
  File "/usr/share/ansible/collections/ansible_collections/cisco/ios/plugins/module_utils/network/ios/ios.py", line 94, in get_capabilities
    capabilities = Connection(module._socket_path).get_capabilities()
  File "/usr/local/lib/python3.8/site-packages/ansible/module_utils/connection.py", line 200, in __rpc__
    raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
fatal: [wlc_campus_ctes415.bancogalicia.com.ar]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "commands": [
                "show running-config | include wlan \\"Invigal Wlan Profile\\" 3 Invigal"
            ],
            "interval": 1,
            "match": "all",
            "provider": null,
            "retries": 10,
            "wait_for": null
        }
    },
    "msg": "ssh connection failed: None of the prompts looked like password prompts: [b'Clave:']"
}
```

**Comment by bcoca** [comment]
~We used sshpass to handle prompts, see the `sshpass_prompt` option to configure 'custom prompts' (also needs a recent version of sshpass installed). https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ssh_connection.html#parameter-  sshpass_prompt~  nvmd, not ssh connection plugin, didn't even realize this is libssh repo ... which I forgot I'm subscribed to.

**Comment by latota007** [comment]
Hi, I changed  ```ansible_network_cli_ssh_type``` to **paramiko** and after 14 days running playbooks the problem doesn´t appear.

**Comment by Qalthos** [comment]
This should be fixed as of ansible-pylibssh 1.0.0 and ansible.netcommon 3.1.0 with the [password_prompt option](https://docs.ansible.com/ansible/latest/collections/ansible/netcommon/libssh_connection.html#parameter-password_prompt).
